src/wui/wui.c

```diff
@@ -54,24 +54,28 @@ static void update_wui_vars(void) {
     osMutexRelease(wui_thread_mutex_id);
 }
 
-static int process_wui_request(char *request) {
-
-    if (strncmp(request, "!cip ", 5) == 0) {
-        uint32_t ip;
-        if (sscanf(request + 5, "%lu", &ip)) {
-            eeprom_set_var(EEVAR_CONNECT_IP4, variant8_ui32(ip));
+static int process_wui_request(wui_cmd_t *request) {
+
+    if (request->lvl == HIGH_LVL_CMD){
+        if (request->high_lvl_cmd == CMD_SET){
+            if (request->var_msk & WUI_VAR_MSK(VAR_CONNECT_IP4)) {
+                uint32_t ip;
+                if (sscanf(request->arg, "%lu", &ip)) {
+                    eeprom_set_var(EEVAR_CONNECT_IP4, variant8_ui32(ip));
+                }
+            } else if (request->var_msk & WUI_VAR_MSK(VAR_CONNECT_TOKEN)) {
+                variant8_t token = variant8_pchar(request->arg, 0, 0);
```

Comment:
i think this variant is not complete? anyway variant is something outside of WUI library

Reply:
You said to do it without the refactorings.. So it's without the refactorings